src/Tasks/GenerateResource.cs

```diff
@@ -995,7 +995,8 @@ private bool TransformResourceFilesUsingResGen(List<ITaskItem> inputsToProcess,
 
             // We need to do a whole lot of work to make sure that we're not overrunning the command line ... UNLESS
             // we're running ResGen 4.0 or later, which supports response files. 
-            if (!_resgenPath.Equals(Path.GetDirectoryName(NativeMethodsShared.GetLongFilePath(ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("resgen.exe", TargetDotNetFrameworkVersion.Version35))), StringComparison.OrdinalIgnoreCase))
+            if (NativeMethodsShared.IsWindows &&
```

Comment:
Ah, I misunderstood your commit message! Mono runs on windows too, so it would be good to explicitly mention that mono's resgen doesn't support response files.
- The condition needs to have a ` && NativeMethodsShared.IsMono &&` too.
- Update the comment to reflect the mono case.
- The same changes would be required in `ResGen.cs` too
- And IMHO, the commit message could be a little clearer, like:

=============

Fix command line length errors with resgen on mono/osx

Mono's `resgen` tool doesn't support response files. So, response files are
supported only when using .NET's `resgen` (4.0) tool.

This manifested as command line length errors for resgen, on mono/osx (see issue #1776).

```
/Library/Frameworks/Mono.framework/Versions/4.8.0/lib/mono/msbuild/15.0/bin/Microsoft.Common.CurrentVersion.targets(2881,5): error MSB3455: ResGen.exe may not run because the command line is 48351 characters long, which exceeds the maximum length of the command. To fix this problem, please either (1) remove unnecessary assembly references, or (2) make the paths to those references shorter. [/Users/filipnavara/Desktop/emclient/MailClient/MailClient.csproj
```

GenerateResource task only checked for `resgen.exe` path and on OSX, incorrectly
assumed that it must be `resgen 4.0` with support for response files. It 
invokes `ResGen` task, which has an additional condition `IsWindows` for using
response files, triggering the alternative code path but that fails due to command line 
length being too long.
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   So, the `GenerateResource` and `ResGen` tasks should use response files only with
.NET's resgen tool.